packages/nextjs/components/Header.tsx

```diff
@@ -20,14 +56,65 @@ const NavLink = ({ href, children }: { href: string; children: React.ReactNode }
   );
 };
 
+const NavLinks = ({ setIsOpen }: { setIsOpen: React.Dispatch<React.SetStateAction<boolean>> }) => {
+  return (
+    <>
+      {menuItems.map((item, index) => {
+        return (
+          <li key={index}>
+            <NavLink href={item.link} setIsOpen={setIsOpen}>
+              {item.icon}
+              {item.label}
+            </NavLink>
+          </li>
+        );
+      })}
+    </>
+  );
+};
+
+/* End of possible Navlinks Components */
+
 /**
  * Site header
  */
 export default function Header() {
+  const [isOpen, setIsOpen] = useState(Boolean);
+  const ref = useRef<HTMLInputElement>(null);
+
+  // Is this considered accessing the DOM? Is this okay?
+  useEffect(() => {
+    function handler(event: { target: any }) {
+      if (!ref.current?.contains(event.target)) {
+        setIsOpen(false);
+      }
+    }
+    window.addEventListener("click", handler);
+    return () => window.removeEventListener("click", handler);
```

Comment:
>  Is this considered accessing the DOM? Is this okay?

I was unable to find a better way, but please fill free to implement it if you find a better approach 🙌

Any particular reason for using `window` here ? 

maybe we use `document.body.addEventListener("click", handler)`  to be more specific since event will hit document first. checkout [this](https://stackoverflow.com/a/59370985) for more info